modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMFormInstanceLocalServiceImpl.java

```diff
@@ -236,6 +237,9 @@ public void deleteFormInstance(DDMFormInstance ddmFormInstance)
 			ResourceConstants.SCOPE_INDIVIDUAL,
 			ddmFormInstance.getFormInstanceId());
 
+		_ddmFormInstanceReportLocalService.deleteByFormInstanceId(
+			ddmFormInstance.getFormInstanceId());
+
```

Comment:
You are deleting the ddmFormInstanceReport here, and the ddmFormInstanceRecord and ddmFormInstanceRecordVersion will be deleted next.
I'm afraid if an exception will be thrown because there will no longer exist a ddmFormInstanceRecord in [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/model/listener/DDMFormInstanceRecordModelListener.java#L42-L45) :thinking:

Reply:
I noticed later that there is a catch in DDMFormInstanceRecordModelListener.onBeforeRemove() :+1: